### ISSUE TITLE
Cannot select text via either mouse or ctrl-a

### ISSUE THREAD

**Issue by rickyrocker** [TigerVNC/tigervnc#995]
Running CentOS 7 with standard TigerVNC server from CentOS repo. Latest OS patches have been applied the OS is running on VirtualBox 6.1.6   (all x64)

Installed packages showing are 1.8.0-17.el7

Text selection in gnome has always worked fine but for a while now we have been having problems selecting text via either mouse or keyboard. As soon as either the ctrl-a key is released or the left mouse button is released post selection, the selection "disappears".

Other info:
1. This is happening across a number of identical environments.
2. There is no issue with selecting text on the console after running startx
3. the select issue occurs from both TigerVNC and RealVNC clients

I tried increasing logging as follows: vncserver -Log *:stderr:100 and although a lot of information is displayed when I click ctrl-a, nothing stands out as the problem.

I'm not sure what else to say except that this is driving us completely crazy and we have no idea what to do so any help or pointers would be greatly appreciated.

Please let me know if I can provide any further info - thanks

**Comment by CendioOssman** [comment]
This usually happens when there is some misbehaving clipboard manager involved, either on the client or the server. Could you check if you might have any such programs running?

I would also advise checking different client devices. Preferably with different OS.

And please try the latest TigerVNC release from here. 1.8.0 is rather old and we do have fixes for the clipboard since then.

**Comment by rickyrocker** [comment]
upgrading to 1.10.1 unfortunately didn't help. However, I uninstalled virtual box 6.1.6 guest additions from the virtual machine and the problem went away. ....so it appears that Virtual Box guest additions is messing up the selection. This has been occurring since Virtual Box 6.1.2

**Comment by rickyrocker** [comment]
i will open an issue on Virtual Box in parallel.

**Comment by rickyrocker** [comment]
Issue opened on virtual box:  https://forums.virtualbox.org/viewtopic.php?f=3&t=97718

**Comment by CendioOssman** [comment]
That's odd. I would assume that only affected the local X server in the VM, not VNC.

How are you starting the TigerVNC server? And are you running VirtualBox on the same machine as you're running the VNC client?

**Comment by rickyrocker** [comment]
yip - VNC client running on same machine.
Starts via:  systemctl start vncserver@:1 
containing following commands:
# Clean any existing files in /tmp/.X11-unix environment
ExecStartPre=/bin/sh -c '/usr/bin/vncserver -kill %i;/bin/rm -rf /tmp/.X11-unix > /dev/null 2>&1 || :'
ExecStart=/sbin/runuser -l root -c "/usr/bin/vncserver %i -geometry 1360x768 -nolisten tcp"

**Comment by CendioOssman** [comment]
Could you test running the client on a separate machine and see if the issue remains? I want to pinpoint if the bad interaction is on the client or server side.

**Comment by rickyrocker** [comment]
Hi there - so sorry for delay on this, am working remotely and only have my laptop with me. Also, the VM is only visible to my local machine as running VB localhost adapter with local private subdomain  I'm not sure how easily to set up what you are needing.

Just to double check, you did see that:
1. the problem occurs with both TigerVNC and RealVNC clients
2. when we de-install the VB linux guest additions from the CentOS linux VB guest server the clients are connecting to - the problem goes away (and as soon as I re-install the problem returns)

(am asking in case this could eliminate the client from the problem)

sorry again.

**Comment by rickyrocker** [comment]
OK - I have added a bridged network adapter to the VM and accessed it successfully via RealVNC client from another PC on the LAN.  From the remote PC in VNC:
- I am unable to select text to copy either via mouse or ctrl-a
- Also in Gnome Nautilus (Files) application, I am only able to work with Files using the "Move to..." or "Copy to..." menu options. Neither "Copy" nor "Cut" work at all.

Again, de-installing VB guest additions from the server resolves the problem for the remote VNC client.

I hope this is

**Comment by CendioOssman** [comment]
Great, thanks. Hopefully this means the issue is on the server. Unfortunately we cannot be 100% sure as I assume the VirtualBox UI was still open on the server machine?

Could you try adding `-Log Selection:stderr:100` to the vncserver command line and check what the log file says when you copy something? It should tell us what happens in the server.

**Comment by rickyrocker** [comment]
Virtual Box console is command line - I do not start graphic console

**Comment by rickyrocker** [comment]
Please see above in history for following: I tried increasing logging as follows: vncserver -Log *:stderr:100 and although a lot of information is displayed when I click ctrl-a, nothing stands out as the problem.

...is there anything specific I should be looking for?

**Comment by rickyrocker** [comment]
When trying to do a single selection, I see the following using the -Log Selection:stderr:100 switch....

 Selection:   Requesting STRING for PRIMARY selection
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting STRING for PRIMARY selection
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target STRING, property
              STRING, type STRING)
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting STRING for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting STRING for PRIMARY selection
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting STRING for PRIMARY selection

**Comment by CendioOssman** [comment]
I'm afraid this output is from the old 1.8.0. Could you do the same test with 1.10.1?

**Comment by rickyrocker** [comment]
I have already done so – please look above in the thread.

**Comment by CendioOssman** [comment]
We need the log output, and there is no such output for 1.10.1 in this thread.

**Comment by rickyrocker** [comment]
ah - ok, will do as soon as get a gap

**Comment by CendioOssman** [comment]
Have you had the chance to get that log file?

**Comment by rickyrocker** [comment]
Xvnc TigerVNC 1.10.0 - built Dec 22 2019 12:30:10
Copyright (C) 1999-2019 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.
Underlying X server release 12004000, The X.Org Foundation


Sun May 31 23:28:04 2020
 Selection:   Selection owner change for
              _DBUS_SESSION_BUS_SELECTION_root_3bf2be8b88394c76a4df8be6a54eb6c5
VBoxClient: error: No service specified. Quitting because nothing to do!
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Created selection window
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients

Sun May 31 23:28:05 2020
 Selection:   Selection owner change for WM_S0
 Selection:   Selection owner change for _NET_WM_CM_S0
 Selection:   Selection owner change for @server=ibus

Sun May 31 23:28:06 2020
 Selection:   Selection owner change for _XSETTINGS_S0
 Selection:   Selection owner change for CLIPBOARD_MANAGER

Sun May 31 23:28:07 2020
 Selection:   Selection owner change for _NET_SYSTEM_TRAY_S0
 Selection:   Selection owner change for _NET_DESKTOP_MANAGER_S0

Sun May 31 23:28:50 2020
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (2 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (2 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (5 bytes)
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (5 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (4 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection

Sun May 31 23:30:03 2020
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (3 bytes)
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (4 bytes)
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (4 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (36 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (36 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (36 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (3 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (3 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (45 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (45 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (2 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (2 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (2 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (2 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection

Sun May 31 23:31:18 2020
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for CLIPBOARD (target UTF8_STRING,
              property UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (40 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection

Sun May 31 23:31:47 2020
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (25 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (25 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection

Sun May 31 23:32:17 2020
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (32 bytes)
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (32 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection

Sun May 31 23:32:19 2020
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (1 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection

Sun May 31 23:32:25 2020
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target UTF8_STRING, property
              UTF8_STRING, type UTF8_STRING)
 Selection:   Sending clipboard to clients (29 bytes)
 Selection:   Local clipboard lost, notifying clients
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection owner change for CLIPBOARD
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for CLIPBOARD selection
 Selection:   Selection owner change for PRIMARY
 Selection:   Got clipboard notification, probing for formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Compatible format found, notifying clients
 Selection:   Got request for local clipboard, re-probing formats
 Selection:   Requesting TARGETS for PRIMARY selection
 Selection:   Selection notification for CLIPBOARD (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for CLIPBOARD selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection
 Selection:   Selection notification for PRIMARY (target TARGETS, property
              TARGETS, type ATOM)
 Selection:   Requesting UTF8_STRING for PRIMARY selection

**Comment by rickyrocker** [comment]
Sorry such a long paste above. I did notice that the following occurs a lot:

 Selection: Local clipboard lost, notifying clients

It is probably worth noting that we've recently installed  Virtual Box 6.1.8 Guest additions  and the selection problem persists.

thanks very much for following up on this.

**Comment by CendioOssman** [comment]
Thank you for the log. It shows that there is something on the server side that is aggressively taking over the clipboard. Since uninstalling VirtualBox guest additions helps we must assume it is something included there. I don't quite understand how they get started in a VNC session though...

Unfortunately there isn't much we can do now. You'll need to report this issue to VirtualBox and have them adjust the behaviour of their stuff. Feel free to link to this issue so they can better understand the problem.

**Comment by rickyrocker** [comment]
One last update from me is that the issue is definitely caused by Virtual Box 6.1

We have reverted to VB 6.0.24 and problem has disappeared.

Here is the link to my VB ticket closing observations [https://www.virtualbox.org/ticket/19711#comment:2](url)
